lib/oban.ex

```diff
@@ -647,6 +647,25 @@ defmodule Oban do
     |> Query.retry_job(job_id)
   end
 
+  @doc """
+  Retries all jobs that match on the given queryable. Please note that no matter the used
+  queryable it will never retry available, executing or scheduled jobs.
+  If no queryable is given, all jobs in states where retry is possible will be retried.
+
+  ## Example
+
+      # Retries all jobs with priority 0
+      Oban.Job |> Ecto.Query.where(priority: 0) |> Oban.retry_all_jobs()
+      :ok
+  """
+  @doc since: "2.8.0"
+  @spec retry_all_jobs(name :: atom(), queryable :: Ecto.Queryable.t()) :: :ok
```

Comment:
Since we'll get a count, it seems like something we should return. Rather than a plain `:ok`, how about `{:ok, number_of_retries}` here?